File: sale-process/sales/src/main/java/sales/system/sales/service/VentaService.java
package sales.system.sales.service;

import sales.system.sales.model.Venta;

import java.util.List;
import java.util.Optional;

public interface VentaService {
    List<Venta> obtenerTodas();
    Optional<Venta> obtenerPorId(Long id);
    Venta crear(Venta venta);
    Venta actualizar(Long id, Venta venta);  // Nuevo método para actualizar
    void eliminar(Long id);
}

File: sale-process/sales-app/src/app/services/venta.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Venta } from '../interfaces/venta';

@Injectable({
  providedIn: 'root'
})
export class VentaService {

  private apiUrl = 'http://localhost:8080/ventas';

  constructor(private http: HttpClient) { }

  obtenerTodas(): Observable<Venta[]> {
    return this.http.get<Venta[]>(this.apiUrl);
  }

  obtenerPorId(id: number): Observable<Venta> {
    return this.http.get<Venta>(`${this.apiUrl}/${id}`);
  }

  crear(venta: Venta): Observable<Venta> {
    return this.http.post<Venta>(this.apiUrl, venta);
  }

  actualizar(id: number, venta: Venta): Observable<Venta> {
    return this.http.put<Venta>(`${this.apiUrl}/${id}`, venta);
  }

  eliminar(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
